firefox-ios/Client/Assets/RemoteSettingsData/Update_Remote_Settings.py

```diff
@@ -0,0 +1,61 @@
+import requests
+import filecmp
+import shutil
+import os
+import json
+
+CONFIG_FILE = "./firefox-ios/Client/Assets/RemoteSettingsData/Remote_Settings_Fetch_Config.json"
+GITHUB_ACTIONS_PATH = "./firefox-ios/Client/Assets/RemoteSettingsData/"
+GITHUB_ACTIONS_TMP_PATH = f"{GITHUB_ACTIONS_PATH}tmp/"
+
+def fetch_rules(url):
+    response = requests.get(url)
+    if response.status_code == 200:
+        return response.json()["data"]
+    else:
+        print(f"Failed to fetch rules from {url}: {response.status_code}")
+        return None
+
+def save_tmp_rules(rules, tmp_file_path):
+    with open(tmp_file_path, 'w') as tmp_file:
+        json.dump(rules, tmp_file, indent=4)
+    print(f"Saved temporary rules to {tmp_file_path}")
+
+def update_settings_file(tmp_file_path, target_file_path, name):
+    if os.path.exists(target_file_path):
+        if filecmp.cmp(tmp_file_path, target_file_path):
+            print(f"No changes detected for {name}.")
+            os.remove(tmp_file_path)
+            return False
+        else:
+            os.replace(tmp_file_path, target_file_path)
+            print(f"Updated {target_file_path} with new rules for {name}.")
+            return True
+    else:
+        os.replace(tmp_file_path, target_file_path)
+        print(f"Created new rules file {target_file_path} for {name}.")
+        return True
+
+def main():
+    if not os.path.exists(GITHUB_ACTIONS_TMP_PATH):
+        os.mkdir(GITHUB_ACTIONS_TMP_PATH)
+    
+    with open(CONFIG_FILE, 'r') as config_file:
+        config = json.load(config_file)
+    
+    changes_detected = False
+    
+    for rule in config["rules"]:
+        print(f"Fetching rules for {rule['name']} from {rule['url']}")
+        rules = fetch_rules(rule["url"])
+        if rules:
+            tmp_file_path = os.path.join(GITHUB_ACTIONS_TMP_PATH, os.path.basename(rule["file"]))
+            save_tmp_rules(rules, tmp_file_path)
+            if update_settings_file(tmp_file_path, rule["file"], rule["name"]):
+                changes_detected = True
+    
+    if not changes_detected:
+        print("No changes detected in any rules.")
+
```

Comment:
It won't break the code since we check if the directory exists before creating it, but maybe we should remove the directory after we are done ? Something like `shutil.rmtree(GITHUB_ACTIONS_TMP_PATH)`

Reply:
true, I left it here as I wasn't sure what to do here 🤷🏼 
One question that I had was since its run on github actions, does it spin a new VM everytime an action is run or all share the same VM

Reply:
Good question. So looking at the docs, github actions doesn't cache that directory between runs. And since the files in there won't be committed and not  in the final PR it doesn't really matter if we remove it or not 😄

Reply:
We can leave it be then for any logs during that run!